Sources/Plasma/FeatureLib/pfPython/plPythonConvert.h

```diff
@@ -0,0 +1,186 @@
+/*==LICENSE==*
+
+CyanWorlds.com Engine - MMOG client, server and tools
+Copyright (C) 2011  Cyan Worlds, Inc.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+Additional permissions under GNU GPL version 3 section 7
+
+If you modify this Program, or any covered work, by linking or
+combining it with any of RAD Game Tools Bink SDK, Autodesk 3ds Max SDK,
+NVIDIA PhysX SDK, Microsoft DirectX SDK, OpenSSL library, Independent
+JPEG Group JPEG library, Microsoft Windows Media SDK, or Apple QuickTime SDK
+(or a modified version of those libraries),
+containing parts covered by the terms of the Bink SDK EULA, 3ds Max EULA,
+PhysX SDK EULA, DirectX SDK EULA, OpenSSL and SSLeay licenses, IJG
+JPEG Library README, Windows Media SDK EULA, or QuickTime SDK EULA, the
+licensors of this Program grant you additional
+permission to convey the resulting work. Corresponding Source for a
+non-source form of such a combination shall include the source code for
+the parts of OpenSSL and IJG JPEG Library used as well as that of the covered
+work.
+
+You can contact Cyan Worlds, Inc. by email legal@cyan.com
+ or by snail mail at:
+      Cyan Worlds, Inc.
+      14617 N Newport Hwy
+      Mead, WA   99021
+
+*==LICENSE==*/
+
+#ifndef _plPythonConvert_h_
+#define _plPythonConvert_h_
+
+#include <Python.h>
+#include <string_theory/string>
+#include <type_traits>
+#include <tuple>
+
+#include "pyGlueHelpers.h"
+#include "pyObjectRef.h"
+
+namespace plPython
+{
+    template<typename ArgT>
+    inline PyObject* ConvertFrom(ArgT&& value) = delete;
+
+    inline PyObject* ConvertFrom(bool value)
+    {
+        PyObject* result = value ? Py_True : Py_False;
+        Py_INCREF(result);
+        return result;
+    }
+
+    inline PyObject* ConvertFrom(char value)
+    {
+        return PyUnicode_FromFormat("%c", (int)value);
+    }
+
+    inline PyObject* ConvertFrom(const char* value)
+    {
+        return PyUnicode_FromString(value);
+    }
+
+    inline PyObject* ConvertFrom(double value)
+    {
+        return PyFloat_FromDouble(value);
+    }
+
+    inline PyObject* ConvertFrom(float value)
+    {
+        return PyFloat_FromDouble(value);
+    }
+
+    inline PyObject* ConvertFrom(int8_t value)
```

Comment:
This will cause an issue on systems (read: Solaris) which alias `int8_t` as just `char` instead of `signed char`.  It would probably be better to use the base types directly (`char`, `signed`/`unsigned` `char`, `short`, `int`, `long`, etc), as was done for String Theory

Reply:
Just to check, I resolved this correctly in 69b3a4821e84fe833ac12071d9cac842383c1848, right?

Reply:
Yep, looks good to me